script.module.resolveurl/lib/resolveurl/plugins/vidmoly.py

```diff
@@ -16,22 +16,36 @@
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """
 
-from resolveurl.plugins.__resolve_generic__ import ResolveGeneric
+import re
 from resolveurl.lib import helpers
+from resolveurl import common
+from resolveurl.resolver import ResolveUrl, ResolverError
+from six.moves import urllib_parse
 
 
-class VidMolyResolver(ResolveGeneric):
+class VidMolyResolver(ResolveUrl):
     name = 'VidMoly'
     domains = ['vidmoly.me', 'vidmoly.to', 'vidmoly.net']
+    headers = {'User-Agent': common.RAND_UA}
     pattern = r'(?://|\.)(vidmoly\.(?:me|to|net))/(?:embed-|w/)?([0-9a-zA-Z]+)'
 
-    def get_media_url(self, host, media_id):
-        return helpers.get_media_url(
-            self.get_url(host, media_id),
-            patterns=[r'''sources:\s*\[{file:"(?P<url>[^"]+)'''],
-            result_blacklist=['.mpd'],
-            referer=True
-        )
+    def get_media_url(self, host, media_id, subs=False):
+        web_url = self.get_url(host, media_id)
+        self.headers.update({'Referer': web_url, 'Origin': urllib_parse.urljoin(web_url, '/')})
```

Comment:
Origin should not have a trailing '/' sho add a [:-1]
`'Origin': urllib_parse.urljoin(web_url, '/')[:-1]`

Reply:
Right, done.